### DISCUSSION TITLE
Need help in making authentication with ActiveDirectory server

### DISCUSSION THREAD

**Discussion by danatela** [DirectoryTree/LdapRecord#327] [Q&A]
I am trying to make authentication with ActiveDirectory using ldaprecord-laravel. I followed the documentation and made required changes in files. However, I ended up with only `php artisan ldap:test` working and `php artisan ldap:import ldap` showing that there are no users to import.

When I use [online LDAP test server](https://www.forumsys.com/tutorials/integration-how-to/ldap/online-ldap-test-server/), I can go further and make `Auth::attempt(['uid' => 'einstein', 'password' => 'password'])` in Tinker, but the web login still doesn't work. With AD, I can't auth attempt using neither samaccountname, nor username, nor uid. Though plain auth using ldap_connect and ldap_bind works.

There are few changes in my configuration as compared to documentation, i.e. user_login instead of username, but I made proper methods to handle them.

How can I find out what causes the problem?

**Comment by stevebauman** [comment]
Hi @danatela,

To debug these types of issues, make sure you have logging enabled and try authenticating again. Then check your Laravel application logs inside of the `storage/logs` directory.

You can also use the `php artisan ldap:browse` command to see if you are able to successfully navigate through your configured LDAP server. This will ensure that you have the correct `base_dn` set, as well as an account that has the permission to browse the directory and locate the LDAP users you are attempting to authenticate with.

Hope this helps!

**Reply by danatela** [reply]
Thanks @stevebauman 

The short answer to my question was to replace 

    protected function credentials(Request $request)
    {
        return [
            'uid' => $request->username,
            'password' => $request->password,
        ];
    }
to

    protected function credentials(Request $request)
    {
        return [
            'uid' => $request->user_login,
            'password' => $request->password,
        ];
    }
It was not obvious from the  documentation that I had to do it.

**Reply by stevebauman** [reply]
https://ldaprecord.com/docs/laravel/v2/auth/plain/laravel-ui#login-controller
